python-package/xgboost/data.py

```diff
@@ -909,9 +909,19 @@ def _transform_cudf_df(
     enable_categorical: bool,
 ) -> Tuple[ctypes.c_void_p, list, Optional[FeatureNames], Optional[FeatureTypes]]:
     try:
-        from cudf.api.types import is_categorical_dtype
+        from cudf.api.types import is_bool_dtype, is_categorical_dtype
```

Comment:
Do we have a problem with pandas? We started using the array interface with it recently.

Reply:
I thought we convert boolean columns to float? https://github.com/dmlc/xgboost/blob/6e5c335ceaadf2f4c5b61b115de64ea93dbbe6e6/python-package/xgboost/data.py#L219